va/tlsalpn.go

```diff
@@ -160,26 +156,21 @@ func (va *ValidationAuthorityImpl) getChallengeCert(
 	return certs[0], &cs, nil
 }
 
-// tlsDial does the equivalent of tls.Dial, but obeying a context. Once
-// tls.DialContextWithDialer is available, switch to that.
+// tlsDial does the equivalent of tls.Dial, but obeying a context for both the
+// dial and handshake operations.
 func (va *ValidationAuthorityImpl) tlsDial(ctx context.Context, hostPort string, config *tls.Config) (*tls.Conn, error) {
 	ctx, cancel := context.WithTimeout(ctx, va.singleDialTimeout)
 	defer cancel()
 	dialer := &net.Dialer{}
 	netConn, err := dialer.DialContext(ctx, "tcp", hostPort)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("during dial: %w", err)
```

Comment:
Just to confirm: the "during dial", "during handshake", and "getting IP addresses" parts of these messages do not become user-facing?

Reply:
Correct, it gets replaced with `probs.Connection("Timeout after connect (your server may be slow or overloaded)"` inside `detailedError()`. This was just useful for my debugging, and didn't seem necessary to remove when I was done.